RevenueCatUI/UIKit/PaywallViewController.swift

```diff
@@ -209,17 +211,20 @@ public class PaywallViewController: UIViewController {
 
     private var hostingController: UIHostingController<PaywallContainerView>? {
         willSet {
-            guard let oldValue = self.hostingController else { return }
+            guard let oldController = self.hostingController else { return }
 
-            oldValue.willMove(toParent: nil)
-            oldValue.view.removeFromSuperview()
-            oldValue.removeFromParent()
+            oldController.willMove(toParent: nil)
+            oldController.view.removeFromSuperview()
+            oldController.removeFromParent()
         }
 
         didSet {
             guard let newController = self.hostingController else { return }
 
             self.addChild(newController)
+
+            self.view.subviews.forEach { $0.removeFromSuperview() }
```

Comment:
Just to check but this is still needed right? As in, it may still happen that addding the controller triggers a new set of the controller?

Reply:
I added this just in case. To guarantee we are clearing before adding any subview. In any case, just this line didn't fix it, adding the subview was triggering the load view, and eventually calling `didSet`

- willSet (removes views)
- didSet
	- removes all views
	- start of addSubview
	    - loadView
			- willSet (removes views)
			- didSet
				- remove all views
				- addSubview
	- finish adding subview

At this point there are two views and the removal of all views in didSet didn't fix it.

Reply:
Makes sense thanks!